docs/design-docs/sync.md

```diff
@@ -0,0 +1,60 @@
+# Sync WordPress Studio Sites
+
+## About this doc
+
+This design document describes the requirements, architecture, and data models for the **Studio Sync** feature.
+
+## Context
+
+WordPress Studio Sync enables developers to pull a live site down for local development (including themes, plugins, media, and the database) and then push changes back without manual exports or FTP. It supports both WordPress.com and Pressable sites with a Jetpack connection.
+
+## Terminology
+
+- **Sync**: Replicating files and database content between a local machine and a remote site in any direction.
+- **Push**: Copying changes from a local machine to a remote (staging or production) site.
+- **Pull**: Copying changes from a remote site down to a local machine.
+- **Staging site**: A staging site is hosted on WordPress.com and is connected to its production site. It is used to test the sync feature and serves as the source of truth for this feature. WordPress.com staging sites can sync with production sites and vice versa.
```

Comment:
A few things here:
1. I don't really get the part about "serving as the source of truth for this feature". Users can sync from Studio to both staging sites and production sites, no? In what sense are staging sites the source of truth?
2. Nit, but we just defined "sync" as the thing that Studio does with WordPress.com or Pressable, yet here we say that "WordPress.com staging sites can sync with production sites". It might be worth clarifying that staging site sync is an entirely separate feature.

Reply:
Good catch! Done: 2a6aeb4a44efe521b0af4debd3fb2e95efd5f871